gradio/helpers.py

```diff
@@ -298,9 +341,56 @@ async def load_example(example_id):
             else:
                 client_utils.synchronize_async(self.cache)
 
+    async def _handle_callable_as_generator(self, *args):
+        if self.fn is None:
+            raise ValueError("Cannot lazy-cache examples if no function is provided")
+        if inspect.iscoroutinefunction(self.fn):
+            result = await self.fn(*args)
+            yield result
+        elif inspect.isasyncgenfunction(self.fn):
+            async for item in self.fn(*args):
+                yield item
+        elif inspect.isgeneratorfunction(self.fn):
+            for item in self.fn(*args):
+                yield item
+        else:
+            yield self.fn(*args)
```

Comment:
This will block the event loop

Reply:
Ah right I was going to find a better way to do this. Do you know how to handle this?

Reply:
We need to split this into two functions, one a coroutine and the other a normal function and pass the appropriate one to the event depending on the type of the `fn` parameter

Reply:
There may be other solutions ^

Reply:
Ok let me take a look! Thanks

Reply:
What I suggested abouve is what we do for the iterators in the non-lazy case basically

Reply:
Right, I was also seeing if it made sense to do this: https://github.com/gradio-app/gradio/blob/e0a8b7fa2c60591f7fc6861707f65bf85f4ad27f/gradio/blocks.py#L1266